builtin/k8s/platform.go

```diff
@@ -367,6 +368,23 @@ func (p *Platform) Deploy(
 
 	timeout := 10 * time.Minute
 
+	cancelMu := &sync.Mutex{}
+	pollingComplete := false
```

Comment:
I'm hoping there's a better way to handle this than this mutex. Open to suggestions.